hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -546,13 +542,13 @@ private void registerInvalidationListener() {
             return;
         }
 
-        EventHandler eventHandler = createInvalidationEventHandler();
-        nearCacheMembershipRegistrationId = addNearCacheInvalidationListener(eventHandler);
+        ConnectedServerVersionAwareNearCacheEventHandler handler = new ConnectedServerVersionAwareNearCacheEventHandler();
+        nearCacheMembershipRegistrationId = addNearCacheInvalidationListener(handler);
+        if (!handler.supportsRepairableNearCache) {
```

Comment:
The usage of the `supportsRepairableNearCache` field and method is a bit confusing for me (as in: when is the field initialized, when do I have to use the method instead etc.). But the classes and their hierarchy and lifecycle are really not easy to read on GitHub.

Reply:
fixed